### PR TITLE
Add support for "packaging" monorepo modules

### PR DESCRIPTION
**Pull request by Monkatraz**
It's now trivial to get a module built in such a way that it's publishable on NPM. 

I will copy paste the comment I left in `vite-pack.js`:
```
/*
 * This script allows for the creation of NPM packages from a module in the
 * monorepo. Specifically, it bundles the module and copies the
 * package.json (and friends) into the modules "dist" folder. This is where
 * the package will be published from.
 *
 * This allows for the actual package.json to still point to the module's
 * source, so that the monorepo itself doesn't need to have every module
 * built to work. In general, the monorepo uses no built modules when
 * developing or building websites.
 *
 * Packages are published with their sources, which has two main advantages:
 * 1. The `types` field in the package.json is set to the module's source
 * 2. The package is built with sourcemaps that reference the source
 *
 * The disadvantage is the package is larger, but it's a good tradeoff.
 */
 ```

Running the script:
image
image

Comparing the generated `package.json`:
```jsonc
// original:

{
  "name": "ftml-wasm-worker",
  "license": "agpl-3.0-or-later",
  "description": "A web workerized version of ftml-wasm",
  "version": "0.0.0",
  "keywords": [
    "wikijump"
  ],
  "private": true,
  "scripts": {},
  "type": "module",
  "main": "src/index.ts",
  "eslintConfig": {
    "extends": "../../.eslintrc.js"
  },
  "dependencies": {
    "ftml-wasm": "workspace:*",
    "threads": "^1.6.5",
    "threads-worker-module": "workspace:*",
    "wj-util": "workspace:*"
  }
}

// generated:

{
  "name": "ftml-wasm-worker",
  "license": "agpl-3.0-or-later",
  "description": "A web workerized version of ftml-wasm",
  "version": "0.0.0",
  "keywords": [
    "wikijump"
  ],
  "scripts": {},
  "main": "cjs/ftml-wasm-worker.cjs",
  "dependencies": {
    "ftml-wasm": "^0.0.0",
    "threads": "^1.6.5",
    "threads-worker-module": "^0.0.0",
    "wj-util": "^0.0.0"
  },
  "types": "src/index.ts",
  "module": "esm/ftml-wasm-worker.mjs",
  "exports": {
    ".": {
      "import": "esm/ftml-wasm-worker.mjs",
      "require": "cjs/ftml-wasm-worker.cjs"
    }
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/scpwiki/wikijump.git",
    "directory": "client/modules/ftml-wasm-worker"
  },
  "bugs": "https://scuttle.atlassian.net/servicedesk/customer/portal/2",
  "homepage": "https://github.com/scpwiki/wikijump/tree/develop/client/modules/ftml-wasm-worker"
}
```

As you can see, this script does a lot of stuff in a relatively small amount of code.

### CODE REVIEW

File: client/scripts/vite-config.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      // because esbuild rips out comments, esbuild will not preserve
+      // the comments we need for ignoring lines
+      // however, it does preserve legal comments, /*! or //!
+      // but c8 doesn't recognize those!
+      // so we have to transform those back into normal comments
+      // before we let c8 parse them
+      {
+        transform(code, id) {
+          // use two spaces so we don't change the length of the document
+          code = code.replaceAll("/*! c8", "/*  c8")
```
Comment (emmiegit) [inline client/scripts/vite-config.js:55]: Why does changing the length of the document matter?

**Review by emmiegit — APPROVED** [review]

File: client/scripts/vite-config.js (hunk 1 above)
Comment (Monkatraz) [reply client/scripts/vite-config.js:55]: The source map (literally, just a map from an output file's ranges to a input file's ranges) would be affected for every position past that point.

Comment (emmiegit) [reply client/scripts/vite-config.js:55]: Ah right, forgot that's a thing.

**Comment by Monkatraz** [comment]
thanks @ammongit

**Pull request merged**